examples/go-kafkacat/go-kafkacat.go

```diff
@@ -52,6 +140,10 @@ func runProducer(config *kafka.ConfigMap, topic string, partition int32) {
 		for ev := range drs {
 			m, ok := ev.(*kafka.Message)
 			if !ok {
+				switch e := ev.(type) {
+				case kafka.OAuthBearerTokenRefresh:
+					handleOAuthBearerTokenRefreshEvent(p, e, config)
```

Comment:
I think we'll want a better way to provide the config here so the application does not need to hang on to it.

In librdkafka we should either pass the "sasl.oauthbearer.config" value as an argument to the refresh callback, which is the more direct approach, or provide an API to extract config values from an rd_kafka_t handle, which is the more generic and flexible approach.

I'm voting for the latter and if we agree on that being the solution I'll make the PR for librdkafka.

What do you think?

Reply:
IIUC, you would prefer the call to be simply `handleOAuthBearerTokenRefreshEvent(p, e)` rather than passing in the config explicitly as a third parameter, and then inside that function the implementation could retrieve the `sasl.oauthbearer.config` value via this new API.  If that is a correct understanding, then yes, I think it makes sense.  Now that you mention it, actually, it does seem strange that there is no way to interrogate the underlying configuration.

> I'm voting for the latter and if we agree on that being the solution I'll make the PR for librdkafka.

That would be great

Reply:
Come to think of it, I think we should do both:
 1. pass the value of sasl.oauthbearer.config to the refresh callback, since most implementations will need configuration.
 2. provide a way to query any configuration value, for more complex implementations that perhaps rely on other client settings as well, whatever that is.

Reply:
OK, sounds good.  Will you do the PR to take care of both, or should I do the refresh callback one?

Reply:
All set, the call is now simply `handleOAuthBearerTokenRefreshEvent(p, e)`